fvm/transactionInvoker.go

```diff
@@ -78,10 +78,17 @@ func (i *TransactionInvoker) Process(
 		sth.SetActiveState(parentState)
 	}()
 
-	env, err := NewTransactionEnvironment(*ctx, vm, sth, programs, proc.Transaction, proc.TxIndex, span)
+	env, err := NewTransactionEnvironment(*ctx, vm, sth, programs, proc.Transaction, proc.TxIndex, span, true)
 	if err != nil {
 		return fmt.Errorf("error creating new environment: %w", err)
 	}
+
+	// TODO(patrick): make env reusable on error
+	errEnv, err := NewTransactionEnvironment(*ctx, vm, sth, programs, proc.Transaction, proc.TxIndex, span, false)
```

Comment:
Does it make sense creating this even though we most of the time (assuming successful transactions are prevalent) don't need it?

Not loading config from state should be ok as it does not matter how much we meter during transaction fee deduction. We cannot charge for that part. It does might if we hit some sort of limit, but the deduct fees method on the service account should be trusted not to do any infinite loops and the limiting is disabled right now anyway.

I think this (not loading config from state) is ok, but it does deserve some comments in the code what and why this is the case.

Reply:
(note that this may not matter after #2971)

I'll add a comment to the file.

There's no need to re-initialize the meter weights/limits since the transaction state (sth) is shared.  Creating new env is cheap once we stop re-reading the environment's meter weights/limits.

Reply:
since #2971 is already approved, i'll just merge that first and simplify this (and skip adding a new comment)